compiler/src/label_check.ml

```
@@ -0,0 +1,43 @@
+open Prog 
+open Utils
+
+(**
+  Duplicate label error
+  (L.t * string) : location and function name of the label already defined
+  (string) : name of the function that trigger the error
+*)
+type duplicate_label_error = (L.t * string) * string
+
+let pp_duplicate_label_error fmt ((dup_loc, f_full_name), dup_full_name) =
+  Format.fprintf fmt
+    "Function '%s' (declared at %s) and function '%s' have conflicting labels names."
+    dup_full_name 
+    (Location.tostring dup_loc)
+    f_full_name
+
+exception DuplicateLabelError of L.t * duplicate_label_error
+
+let rs_duplicate_label_error ~loc (duplicate_type_error) =
+  raise (DuplicateLabelError (loc, duplicate_type_error))
+
+let check_labels_fun (f: ('len,'info,'asm) gfunc) (label_map: (L.t * string) Ms.t) = 
+  let fn_label = PrintCommon.escape f.f_name.fn_name in
+  if f.f_cc != Internal then 
+    match Ms.find_opt fn_label label_map with
+    | None -> Ms.add fn_label (f.f_loc ,f.f_name.fn_name) label_map
+    | Some (loc, name) -> rs_duplicate_label_error ~loc:f.f_loc ((loc, name), f.f_name.fn_name)
+  else label_map
+
+(**
+Check if functions have conflicting names when translated to their assembly label names.
+Raise `DuplicateLabelError` if a conflict is found between two non inline functions (because inline function are substituted before assembly traduction).
+*)
+let check_labels (prog: ('len,'info,'asm) gprog) = 
```

This should return the list of conflicts instead of failing early.